src/app/CMakeLists.txt

```diff
@@ -77,31 +77,95 @@ set_source_files_properties(
         MACOSX_BUNDLE ON
         MACOSX_BUNDLE_BUNDLE_NAME "${EXECUTABLE}"
-        MACOSX_BUNDLE_INFO_PLIST ${qBittorrent_BINARY_DIR}/dist/mac/Info.plist
+        MACOSX_BUNDLE_INFO_PLIST "${qBittorrent_BINARY_DIR}/dist/mac/Info.plist"
     )
     target_sources(qbt_app PRIVATE
-        ${QT_TRANSLATIONS}
-        ${qBittorrent_SOURCE_DIR}/dist/mac/qt.conf
-        ${qBittorrent_SOURCE_DIR}/dist/mac/qBitTorrentDocument.icns
-        ${qBittorrent_SOURCE_DIR}/dist/mac/qbittorrent_mac.icns
+        ${QT_TRANSLATIONS_LIST}
+        "${qBittorrent_SOURCE_DIR}/dist/mac/qt.conf"
+        "${qBittorrent_SOURCE_DIR}/dist/mac/qBitTorrentDocument.icns"
+        "${qBittorrent_SOURCE_DIR}/dist/mac/qbittorrent_mac.icns"
     )
 elseif (CMAKE_SYSTEM_NAME STREQUAL "Windows")
     set_target_properties(qbt_app PROPERTIES WIN32_EXECUTABLE ON)
     if (MINGW)
-        target_sources(qbt_app PRIVATE ${qBittorrent_SOURCE_DIR}/src/qbittorrent_mingw.rc)
+        target_sources(qbt_app PRIVATE "${qBittorrent_SOURCE_DIR}/src/qbittorrent_mingw.rc")
     else()
-        target_sources(qbt_app PRIVATE ${qBittorrent_SOURCE_DIR}/src/qbittorrent.rc)
+        target_sources(qbt_app PRIVATE "${qBittorrent_SOURCE_DIR}/src/qbittorrent.rc")
+    endif()
+    target_sources(qbt_app PRIVATE "${qBittorrent_SOURCE_DIR}/src/qbittorrent.exe.manifest")
+
+    # Copy DLLs to the build folder after the build, so that the executable can be run from there.
+    # When using vcpkg, everything is automatically copied except for some stuff handled by windeployqt;
+    # this is no problem because the copy_if_different command is idempotent in this case
+    # TODO: works for MSVC, what about MinGW?
+    # TODO: This can probably be greatly simplified by using TARGET_RUNTIME_DLLS instead, in CMake >= 3.21
+    if (MSVC_RUNTIME_DYNAMIC)
+        # Must manually find these DLLs, because the target names for these libs point at the .lib files, not the .dll files
+        # Handling zlib is trickier than OpenSSL because FindZLIB sucks
+        cmake_path(SET OPENSSL_DLLS_PATH "${OPENSSL_CRYPTO_LIBRARY}")
+        cmake_path(GET OPENSSL_DLLS_PATH PARENT_PATH OPENSSL_DLLS_PATH)
+        cmake_path(SET OPENSSL_DLLS_PATH NORMALIZE "${OPENSSL_DLLS_PATH}/../bin")
+        find_file(OPENSSL_LIBSSL_DLL "libssl-1_1-x64.dll" PATHS "${OPENSSL_DLLS_PATH}" DOC "OpenSSL::SSL DLL path" REQUIRED
+            NO_DEFAULT_PATH NO_PACKAGE_ROOT_PATH NO_CMAKE_PATH NO_CMAKE_ENVIRONMENT_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH
+        )
+        find_file(OPENSSL_LIBCRYPTO_DLL "libcrypto-1_1-x64.dll" PATHS "${OPENSSL_DLLS_PATH}" DOC "OpenSSL::Crypto DLL path" REQUIRED
+            NO_DEFAULT_PATH NO_PACKAGE_ROOT_PATH NO_CMAKE_PATH NO_CMAKE_ENVIRONMENT_PATH NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH
+        )
+        list(APPEND ZLIB_NAMES_LIST "z.dll" "zlib.dll" "zdll.dll" "zlib1.dll" "zd.dll" "zlibd.dll" "zdlld.dll" "zlibd1.dll" "zlib1d.dll")
```

Comment:
I'm pretty sure CMake knows the exact name of the \.dll file, so you can avoid this kind of guessing.

Reply:
@glassez 

Unfortunately, I don't think that's the case. `FindZLIB` doesn't give you much to work with. You can experiment and see the variables/properties it sets. I couldn't find any other robust way other than copying the "name guessing" that `FindZLIB` itself does. This a problem unique to `zlib`, which for historical reasons can have all these different names. You can see that there is no such problem with OpenSSL, since the DLL name is "standard" (unless you also allow really old versions that might have the old `eay` names).

Also, rebased to fix conflicts.